tests/integration/test_charm.py

```diff
@@ -46,3 +56,48 @@ async def test_build_and_deploy(ops_test: OpsTest):
             timeout=1000,
         )
         assert ops_test.model.applications[APP_NAME].units[0].workload_status == "active"
+
+
+async def test_ingress_relation(ops_test: OpsTest):
+    await ops_test.model.deploy(
+        TRAEFIK,
+        application_name=TRAEFIK_PUBLIC_APP,
+        channel="latest/edge",
+        config={"external_hostname": "some_hostname"},
+    )
+    await ops_test.model.deploy(
+        TRAEFIK,
+        application_name=TRAEFIK_ADMIN_APP,
+        channel="latest/edge",
+        config={"external_hostname": "some_hostname"},
+    )
+    await ops_test.model.add_relation(f"{APP_NAME}:admin-ingress", TRAEFIK_ADMIN_APP)
+    await ops_test.model.add_relation(f"{APP_NAME}:public-ingress", TRAEFIK_PUBLIC_APP)
+
+    await ops_test.model.wait_for_idle(
+        apps=[TRAEFIK_PUBLIC_APP, TRAEFIK_ADMIN_APP],
+        status="active",
+        raise_on_blocked=True,
+        timeout=1000,
+    )
+
+
+async def test_has_public_ingress(ops_test: OpsTest):
+    # Get the traefik address and try to reach hydra
+    public_address = await get_unit_address(ops_test, TRAEFIK_PUBLIC_APP, 0)
+
+    resp = requests.get(
+        f"http://{public_address}/{ops_test.model.name}-{APP_NAME}/.well-known/jwks.json"
+    )
+
+    assert resp.status_code == 200
+
+
+async def test_has_admin_ingress(ops_test: OpsTest):
+    # Get the traefik address and try to reach hydra
+    admin_address = await get_unit_address(ops_test, TRAEFIK_ADMIN_APP, 0)
+
+    resp = requests.get(f"http://{admin_address}/{ops_test.model.name}-{APP_NAME}/admin/clients")
+
+    assert resp.status_code == 200
+    assert resp.json() == []
```

Comment:
Maybe we could make a post request to create a test client before getting them, otherwise I'd remove this assertion as it won't add much to the test